### ISSUE TITLE
Deploying Leafmap Streamlit App on HuggingFace Space (using localtileserver)

### ISSUE THREAD

**Issue by jordancaraballo** [opengeos/leafmap#738]
### Environment Information

-   leafmap version: latest
-   Python version: 3.10
-   Operating System: Docker container, packages listed below

```bash
requirements.txt

--find-links=https://girder.github.io/large_image_wheels GDAL
geopandas
leafmap
owslib
rioxarray
streamlit-extras
psutil
localtileserver
cartopy
huggingface_hub
```

### Description

Deploying Leafmap Streamlit App on HuggingFace Space (using localtileserver) produces an error when trying to check the underlying system. I believe that this is failing at the leafmap side when checking for the environment. This works very well on streamlit cloud, with the exception of rasters not showing because of them (streamlit cloud) lacking support for localtileserver.

When deployed to huggingface, these rasters are locally downloaded to the space storage from HugginFace datasets, and then visualized. This works locally on a laptop, so my assumption is this is related to the way of capturing the running service.

Also documented the issue here https://github.com/loboda-umd/fire-hnl-streamlit/issues/2. 

### What I Did

Software is available here https://github.com/loboda-umd/fire-hnl-streamlit, which is consumed by https://huggingface.co/spaces/loboda-umd/fire-hnl-streamlit. But the core elements of the app are listed below.

```
import os

# should not matter since this is not JH based
os.environ['LOCALTILESERVER_CLIENT_PREFIX'] = 'proxy/{port}'

import streamlit as st
import leafmap.foliumap as leafmap
from glob import glob
from pathlib import Path
from datetime import date, timedelta
from huggingface_hub import hf_hub_download, snapshot_download

st.set_page_config(layout="wide")

markdown = """
FiRE-HNL Beta Dashboard
Validation of lightning probability maps during fire season 2024.
"""

st.sidebar.title("About")
st.sidebar.info(markdown)
logo = "https://i.imgur.com/UbOXYAU.png"
st.sidebar.image(logo)

dataset_path_2016 = snapshot_download(
    repo_id="loboda-umd/fire-hnl-lightning-2016",
    repo_type="dataset",
)

dataset_path_2024 = snapshot_download(
    repo_id="loboda-umd/fire-hnl-lightning-2024",
    repo_type="dataset",
)

def format_filename(filename):
    return f'{Path(filename).stem}'.split('_')[1]

# define layers dict
options_2024 = sorted(glob(os.path.join(dataset_path_2024, '*/*.tif')))
options_2016 = sorted(glob(os.path.join(dataset_path_2016, '*/*.tif')))

st.title(f"Lightning Forecast: {format_filename(options_2024[0])} to {format_filename(options_2024[-1])}")

col1, col2 = st.columns([4, 1])
index = 0

with col2:
    current_forecast_date = st.selectbox("Current Forecast Window:", options_2024, index, format_func=format_filename)
    previous_forecast_date = st.selectbox("Previous Forecast Window:", options_2024, index, format_func=format_filename)
    legacy_forecast_date = st.selectbox("Legacy Examples:", options_2016, index, format_func=format_filename)

m = leafmap.Map(
    locate_control=True, latlon_control=True,
    draw_export=True, minimap_control=True,
    center=[30, -40],
    zoom=13

)
m.add_basemap("OpenTopoMap")

with col1:

    m.add_raster(
        current_forecast_date,
        bands=[1],
        vmin=0,
        vmax=100,
        colormap='jet',
        layer_name=Path(current_forecast_date).stem
    )

    # folium does not support this
    # m.add_time_slider(layers_dict, time_interval=1)
    m.to_streamlit(height=700)
```

The main error occurs when trying to `add_raster`:

```bash
2024-05-12 14:26:39.647 Uncaught app exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 600, in _run_script
    exec(code, module.__dict__)
  File "/home/user/app/pages/1_⚡_Alaska_Lightning.py", line 79, in <module>
    m.add_raster(
  File "/usr/local/lib/python3.10/site-packages/leafmap/foliumap.py", line 551, in add_raster
    tile_layer, tile_client = get_local_tile_layer(
  File "/usr/local/lib/python3.10/site-packages/leafmap/common.py", line 2886, in get_local_tile_layer
    if is_studio_lab():
  File "/usr/local/lib/python3.10/site-packages/leafmap/common.py", line 6481, in is_studio_lab
    output = psutil.Process().parent().cmdline()
AttributeError: 'NoneType' object has no attribute 'cmdline'
```

My suspicion is that this is related to the following function trying to capture the underlying environment.

```bash
under leafmap/common.py

def is_studio_lab():
    """Check if the current notebook is running on Studio Lab.

    Returns:
        bool: True if the notebook is running on Studio Lab.
    """

    import psutil

    output = psutil.Process().parent().cmdline()

    on_studio_lab = False
    for item in output:
        if "studiolab/bin" in item:
            on_studio_lab = True
    return on_studio_lab
```

**Comment by giswqs** [comment]
@jordancaraballo Thank you for reporting and providing the debugging info. That's very helpful. I have fixed it in #739. Will release a new version shortly.

**Comment by jordancaraballo** [comment]
@giswqs Thanks for the quick fix, it bypassed the studio problem. Unfortunately, the line that follows it is the AWS check, which follows the same pattern. Besides this, I think all others should be fine. 

```python
elif is_on_aws():
    os.environ["LOCALTILESERVER_CLIENT_PREFIX"] = "proxy/{port}"
```

Hopefully this is the last hurdle to get this portion working on HuggingFace. Hopefully this is not a similar situation compared to our streamlit cloud counterpart.
